### DISCUSSION TITLE
No data found at `state.TestApi`. Did you forget to add the reducer to the store?

### DISCUSSION THREAD

**Discussion by rolanday** [reduxjs/redux-toolkit#4001] [Q&A]
Hello,

I'm getting the following console error when I add an RTK Query API to my store:

```
No data found at `state.testApi`. Did you forget to add the reducer to the store?
```

The api is definitely added to the store (see below) and the query data is successfully showing in "RTK Query" tab in Redux DevTools, and in rendering in the app via data from use the use hook -- i.e., so functional, everything is working.

I'm running Remix, so I set provider serverState for redux state (e.g., user, settings) so I don't get hydration errors. But problem is:

1. If don't know how to set server state for RTK Query API,  or if even makes send to do so, but TS complains  if not set.
2. If I remove serverState altogether, then no more  console error telling me no data found, but I'm back to having hydration errors (so not an option, but mentioning because hints at source of problem, I think :))

Any ideas? Thanks!

```
// store.ts
export const store = configureStore({
  reducer: {
    [testApi.reducerPath]: testApi.reducer,
    user: userReducer,
    settings: settingsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware()
      .concat(testApi.middleware),
});
```

**Comment by rolanday** [comment]
Hello, I'm closing this out because, while very much like the rtk data model, I've decided to switch back to using react-query & jotai because it's my perception that redux is best suited for traditional spas, and ssr is an afterthought (or not a priority due to limited resources). Thanks again for all the great work on this toolkit and congrats on 2.0!!

**Reply by markerikson** [reply]
It's also the holidays, and we're dealing with a mixture of travel and family stuff right now :)

Without an example project to look into, I don't know what might be happening here based on that short of a description.

That said, tbh I've never even tried to use RTK with Remix at all, and I haven't seen anyone else try either.

**Reply by rolanday** [reply]
Thanks Mark, you owe me no explanation -- super impressed with how active you are championing RTK and helping the community. Re: Remix, mostly works great but, like said, I needed to jump through some SSR hoops and last issue I ran into had me starting to feel like pushing a square peg through a round hole. I mainly use backend for all but the initial page load , which is why rtk could have made sense for me (i.e., not typical remix use-case), and the latest remix release just added some nice BFF features so that was also another factor. Thanks again and best wishes to you this holiday season!

**Comment by EuclidesJoao** [comment]
![Captura de tela 2024-08-27 103702](https://github.com/user-attachments/assets/d1122af1-61bb-4031-b20d-61bd01c498b8)
I dealing with the same issue 

THE STORE CODE: 
```js
/* eslint-disable prettier/prettier */
import { configureStore } from '@reduxjs/toolkit'
import disciplinaReducer from '../features/disciplina/disciplinaSlice'
import turmaReducer from '../features/turma/turmaSlice'
import menubarReducer from '../features/menubar/menubarSlice'
import { turmaApi } from '../features/turma/turmaApi'
import { disciplinaAPI } from "../features/disciplina/disciplinaApi";
import { salaAPI } from "../features/sala/salaAPI";
import { cursoAPI } from "../features/curso/cursoAPI";
import { classeAPI } from "../features/classe/classeAPI";
import { categoriaCursoAPI } from "../features/categoria_curso/categoria_cursoAPI";
import cursoReducer from "../features/curso/cursoSlice";
import salaReducer from "../features/sala/salaSlice";
import classeReducer from "../features/classe/classeSlice";
import { ano_lectivoAPI } from '../features/ano_lectivo/ano_lectivoAPI'
import { paisesAPI } from '../features/paises'
import { provinciasAPI } from '../features/provincias/provincia.API'
import { municipiosAPI } from '../features/municipios'
import { matriculaAPI } from '../features/matricula/matricula.API'
import { inscricaoAPI } from '../features/inscricao/inscricao.API'
import { professorAPI } from '../features/professor/professor.API'
import { utilizadorAPI } from '../features/utilizador/utilizador.API'
import { comportamentoAPI } from '../features/comportamento/comportamento.API'
import { profileAPI } from '../features/profile/profile.API'

export const store = configureStore({
    reducer:{
        sala: salaReducer,
        curso: cursoReducer,
        classe: classeReducer,
        turma: turmaReducer,
        menuBar: menubarReducer,
        disciplina: disciplinaReducer,
        ano_lectivo: ano_lectivoAPI.reducer,
        
        [ turmaApi.reducerPath ]            : turmaApi.reducer,
        [ disciplinaAPI.reducerPath ]       : disciplinaAPI.reducer,
        [ salaAPI.reducerPath ]             : salaAPI.reducer,
        [ cursoAPI.reducerPath ]            : cursoAPI.reducer,
        [ classeAPI.reducerPath ]           : classeAPI.reducer,
        [ categoriaCursoAPI.reducerPath ]   : categoriaCursoAPI.reducer,
        [ paisesAPI.reducerPath ]           : paisesAPI.reducer,
        [ provinciasAPI.reducerPath ]       : provinciasAPI.reducer,
        [ municipiosAPI.reducerPath ]       : municipiosAPI.reducer,  
        [ matriculaAPI.reducerPath ]        : matriculaAPI.reducer,
        [ inscricaoAPI.reducerPath]         : inscricaoAPI.reducer,
        [ professorAPI.reducerPath]         : professorAPI.reducer,
        [ utilizadorAPI.reducerPath]        : utilizadorAPI.render,
        [ comportamentoAPI.reducerPath]     : comportamentoAPI.render,
        [ profileAPI.reducerPath]           : profileAPI.render
    },
    middleware: (getDefaultMiddleware)=> getDefaultMiddleware()
    .concat([ 
        turmaApi.middleware, 
        disciplinaAPI.middleware, 
        salaAPI.middleware, 
        cursoAPI.middleware,
        classeAPI.middleware,
        categoriaCursoAPI.middleware,
        ano_lectivoAPI.middleware,
        paisesAPI.middleware,
        provinciasAPI.middleware,
        municipiosAPI.middleware,
        matriculaAPI.middleware,
        inscricaoAPI.middleware,
        professorAPI.middleware,
        utilizadorAPI.middleware,
        comportamentoAPI.middleware,
        profileAPI.middleware,
    ])
})
```

**Reply by phryneas** [reply]
Why do you create more than one API?
We recommend to usually have one per application. If you have more than one (again, not recommended), you *have* to specify `reducerPath` during the intialization of each one of those.

**Reply by phryneas** [reply]
Also, why are half of those `.render` and not `.reducer`?
